contracts/ReputationMiningCycle.sol

```diff
@@ -229,14 +237,15 @@ contract ReputationMiningCycle is ReputationMiningCycleStorage, PatriciaTreeProo
       // this is the slot after the last entry, and so our opponentIdx will be the last entry
       // We just move the opponent on, and nothing else happens.
 
-      if (round == 0) {
-        // If we're in round zero, require that no more submissions can be made
-        require(submissionWindowClosed(), "colony-reputation-mining-submission-window-still-open");
-      } else {
-        // Otherwise, ensure that the previous round is complete, and this entry wouldn't possibly get an opponent later on.
+      // In all cases, if the window is still open, the submission could still get an opponent
+      require(submissionWindowClosed(), "colony-reputation-mining-submission-window-still-open");
```

Comment:
We do need the call in `challengeRoundComplete` - it's a public function, and is used by the miners to work out when they can claim a bye by calling `invalidateHash`. I guess it could be reworked as `allPreviousRoundsComplete` or something to work around the double-call.

Reply:
Ok, I don't feel strongly. If you think reworking it is fundamentally better then go for it, otherwise don't worry.